traits/trait_notifiers.py

```diff
@@ -44,7 +44,7 @@ def set_ui_handler(handler):
 
 
 def ui_dispatch(handler, *args, **kw):
-    if threading.current_thread().ident == ui_thread:
+    if threading.current_thread().ident == threading.main_thread().ident:
```

Comment:
No need for the `.ident` here: you can just compare the thread objects directly.
```suggestion
    if threading.current_thread() == threading.main_thread():
```

Reply:
Thanks for the suggestion. Are directly comparing threads vs comparing their ids completely equivalent?

Reply:
I think so, yes. The `==` check is essentially an `is` check (there's no `__eq__` special method created), and there should be exactly one `Thread` object per OS thread.